tests/scenarios/v2-addon-dev-test.ts

```diff
@@ -0,0 +1,185 @@
+          demo: {
+            'button.hbs': `
+              <button {{on 'click' @onClick}}>
+                flip
+              </button>
+            `,
+            'out.hbs': `
+              <out>{{yield}}</out>
+            `,
+            'index.js': `
+                import Component from '@glimmer/component';
+
+                import FlipButton from './button';
+                import BlahButton from './button.hbs';
+                import Out from './out';
+
+                export default class ExampleComponent extends Component {
+                  Button = FlipButton;
+                  Button2 = BlahButton;
+                  Out = Out;
+
+                  @tracked active = false;
+
+                  flip = () => (this.active = !this.active);
+                }
+              `,
+            'index.hbs': `
+              Hello there!
+
+              <this.Out>{{this.active}}</this.Out>
+
+              <this.Button @onClick={{this.flip}} />
+              <this.Button2 @onClick={{this.flip}} />
+            `,
+          },
+        },
+      },
+    });
+
+    addon.linkDependency('@embroider/addon-shim', { baseDir: __dirname });
+    addon.linkDependency('@embroider/addon-dev', { baseDir: __dirname });
+    addon.linkDevDependency('@babel/core', { baseDir: __dirname });
+    addon.linkDevDependency('@babel/plugin-proposal-class-properties', { baseDir: __dirname });
+    addon.linkDevDependency('@babel/plugin-proposal-decorators', { baseDir: __dirname });
+    addon.linkDevDependency('@babel/preset-env', { baseDir: __dirname });
+    addon.linkDevDependency('@rollup/plugin-babel', { baseDir: __dirname });
+    addon.linkDevDependency('rollup', { baseDir: __dirname });
+
+    project.addDevDependency(addon);
+
+    merge(project.files, {
+      tests: {
+        'the-test.js': `
+          import { click, render } from '@ember/test-helpers';
+          import { hbs } from 'ember-cli-htmlbars';
+          import { module, test } from 'qunit';
+          import { setupRenderingTest } from 'ember-qunit';
+
+          module('v2 addon tests', function (hooks) {
+            setupRenderingTest(hooks);
+
+            test('<Demo />', async function (assert) {
+              await render(hbs\`<Demo />\`);
+
+              assert.dom('out').containsText('false');
+
+              await click('button');
+
+              assert.dom('out').containsText('true');
+            });
+
+            test('<Demo::Out />', async function (assert) {
+              await render(hbs\`<Demo::Out>hi</Demo::Out>\`);
+
+              assert.dom('out').containsText('hi');
+            });
+          });
+        `,
+      },
+    });
+  })
+  .forEachScenario(scenario => {
+    Qmodule(scenario.name, function (hooks) {
+      let app: PreparedApp;
+      hooks.before(async assert => {
+        app = await scenario.prepare();
+
+        await execa(path.join(addon.baseDir, 'node_modules/rollup/dist/bin/rollup'), ['-c', './rollup.config.mjs'], {
```

Comment:
I think `addon.baseDir` here is probably messing you up. `forEachScenario` is going to run many times and each one gets a different set of input files, probably in entirely different temp directories. Trying to share the `addon` across all of them isn't likely to work.

This test needs a new feature that no test has needed yet: it wants to run a command in a directory that is not the app root, but instead the addon root. Normally tests can use `app.execute` and that isolates them from needing to worry about most of this. 

I think we should extend PreparedApp so you can conveniently execute in dependencies:

```js
await app.inDependendency('v2-addon').execute('yarn rollup -c ./rollup.config.mjs')
```

Short of having that feature, you can adjust the PreparedApp's dir instead:

```js
app.dir = dirname(require.resolve('v2-addon/package.json', { paths: [app.dir] }))
await app.execute('yarn rollup -c ./rollup.config.mjs');
```

Reply:
why set app.dir? that feels incorrect because the app is the test app? 
I'm confused

Reply:
I don't think the app.dir and execute trick actually builds the files. my package.json test now fails with that code.

Reply:
Opened: https://github.com/ef4/scenario-tester/issues/5